packages/features/bookings/lib/handleSeats.ts

```diff
@@ -0,0 +1,796 @@
+      if (seatAttendee?.id && bookingSeat?.id) {
+        await Promise.all([
+          await prisma.attendee.update({
+            where: {
+              id: seatAttendee.id,
+            },
+            data: {
+              bookingId: newTimeSlotBooking.id,
+            },
+          }),
+          await prisma.bookingSeat.update({
+            where: {
+              id: bookingSeat.id,
+            },
+            data: {
+              bookingId: newTimeSlotBooking.id,
+            },
+          }),
+        ]);
+      }
+
+      const copyEvent = cloneDeep(evt);
+
+      const updateManager = await eventManager.reschedule(copyEvent, rescheduleUid, newTimeSlotBooking.id);
+
+      const results = updateManager.results;
+
+      const calendarResult = results.find((result) => result.type.includes("_calendar"));
+
+      evt.iCalUID = Array.isArray(calendarResult?.updatedEvent)
+        ? calendarResult?.updatedEvent[0]?.iCalUID
+        : calendarResult?.updatedEvent?.iCalUID || undefined;
+
+      await sendRescheduledSeatEmail(copyEvent, seatAttendee as Person);
+      const filteredAttendees = originalRescheduledBooking?.attendees.filter((attendee) => {
+        return attendee.email !== bookerEmail;
+      });
+      await lastAttendeeDeleteBooking(originalRescheduledBooking, filteredAttendees, originalBookingEvt);
+
+      const foundBooking = await findBookingQuery(newTimeSlotBooking.id);
+
+      resultBooking = { ...foundBooking, seatReferenceUid: bookingSeat?.referenceUid };
+    }
+  } else {
+    // Need to add translation for attendees to pass type checks. Since these values are never written to the db we can just use the new attendee language
+    const bookingAttendees = booking.attendees.map((attendee) => {
+      return { ...attendee, language: { translate: tAttendees, locale: attendeeLanguage ?? "en" } };
+    });
+
+    evt = { ...evt, attendees: [...bookingAttendees, invitee[0]] };
+
+    if (eventType.seatsPerTimeSlot && eventType.seatsPerTimeSlot <= booking.attendees.length) {
+      throw new HttpError({ statusCode: 409, message: "Booking seats are full" });
+    }
+
+    const videoCallReference = booking.references.find((reference) => reference.type.includes("_video"));
+
+    if (videoCallReference) {
+      evt.videoCallData = {
+        type: videoCallReference.type,
+        id: videoCallReference.meetingId,
+        password: videoCallReference?.meetingPassword,
+        url: videoCallReference.meetingUrl,
+      };
+    }
+
+    const attendeeUniqueId = uuid();
+
+    await prisma.booking.update({
+      where: {
+        uid: reqBookingUid,
+      },
+      include: {
+        attendees: true,
+      },
+      data: {
+        attendees: {
+          create: {
+            email: invitee[0].email,
+            name: invitee[0].name,
+            timeZone: invitee[0].timeZone,
+            locale: invitee[0].language.locale,
```

Comment:
Access `invitee[0]` only once outside the update call, it'll be cleaner.

Reply:
Addressed in c6e5853e2cf91b9f3adb8dd3da859b2513b86a27